packages/provider-elasticsearch/src/utils/dateUtil.js

```diff
@@ -35,13 +35,13 @@ export let rangeToDatemath = {
   last30Days: { from: 'now-30d', to: 'now' },
   last90Days: { from: 'now-90d', to: 'now' },
   last180Days: { from: 'now-180d', to: 'now' },
-  last12Months: { from: 'now/d-12M', to: 'now' },
-  last15Months: { from: 'now/d-15M', to: 'now' },
-  last18Months: { from: 'now/d-18M', to: 'now' },
-  last24Months: { from: 'now/d-24M', to: 'now' },
-  last36Months: { from: 'now/d-36M', to: 'now' },
-  last48Months: { from: 'now/d-48M', to: 'now' },
-  last60Months: { from: 'now/d-60M', to: 'now' },
+  last12Months: { from: 'now/d-12M', to: 'now/d' },
```

Comment:
If the `to` date is not supposed to be inclusive, should we be doing the following https://github.com/smartprocure/contexture/blob/main/packages/provider-elasticsearch/src/utils/dateUtil.js#L59 It seems like we are somewhat inconsistent on what each `lastX` uses as the `to` expression

Reply:
I'm actually thinking the following might be the best - `now+1/d-1ms`. It should cache nicely while also returning results from the current day. The cache expires after six hours, so each time there is a cache miss the query will return any new records. Thoughts?

Reply:
> I'm actually thinking the following might be the best - `now+1d/d-1ms`. It should cache nicely while also returning results from the current day. The cache expires after six hours, so each time there is a cache miss the query will return any new records. Thoughts?

It makes more sense to me personally to include the current date in the `lastX` expressions, should we adjust the `from` to also do `now+1d/d` in that case so that we are not picking up an extra day in the beginning if the end date include the current day?

Reply:
Elastic datemath [still gets converted to a timestamp](https://github.com/smartprocure/contexture/blob/main/packages/provider-elasticsearch/src/example-types/filters/date.js#L20-L21) so the cache is ineffective

Here's the elastic query for the default issued date filter in spending

<img width="356" alt="Screenshot 2024-06-17 at 3 54 00 PM" src="https://github.com/smartprocure/contexture/assets/4336260/5c17d577-e33c-4bb3-a616-e5889e81da0d">

Reply:
@dshishkov In a meeting with Bradley we decided to be inclusive on the entire current day for either past/future ranges. This means that `to` will always be `now+1d/d-1ms` for past ranges and `from` will always be `now/d` for future ranges.